### DISCUSSION TITLE
Seberapa penting docker untuk dipelajari ?

### DISCUSSION THREAD

**Discussion by teguhkrniawan** [deaafrizal/forum#17] [The Lounge]
Halo semua, nanya dong.

1. Menurut kalian seberapa penting docker untuk dipelajari ?
2. Studi kasus yang bagaimana kita cocok menggunakan docker ?
3. Apakah app yang menggunakan docker sulit untuk di maintenance ketika production ? 
4. Apabila hanya menggunakan 1 teknologi seperti laravel saja, prefer kemana ya, setup server pakai nginx dkk langsung diubuntu nya atau via docker ?

**Comment by codev911** [comment]
1. gak ada ilmu yang gk penting untuk dipelajari, justru pertanyaannya bergunakah atau terimplementasikankah ilmu yg kita pelajari ?
2. Sering latihan dan praktek
3. Justru mudah, karena lebih clean config daripada native
4. itu suka - suka developernya sih

**Reply by teguhkrniawan** [reply]
untuk poin nomor 2, latihan dan prakteknya tentang apa ya mas ?

**Comment by deaafrizal** [comment]
docker bakal penting pada waktunya (setelah mengenal lebih dalam). 
kalau belum coba bakal susah ngebayangin yang namanya isolasi, scalability dan lain2. 

salah 1 contoh kasus yg banyak terjadi: 
ketika lu kerja sama team dengan perangkat OS yang berbeda, setup nya akan berbeda dan malah jadi menghambat waktu development, entah harus install WSL dulu buat developer yg windows, entah harus install apa dlu buat developer yg pake macOS, dan seterusnya. docker seengganya bikin hal gituan jadi lebih konsisten untuk setup apps nya.

ini baru dari sisi kenyamanan developer. belum dari apps  levelnya, service level dan lain2 😄 seru kan

**Reply by teguhkrniawan** [reply]
ohh begitu, menarik bang. Tapi akukan pernah coba pakai docker ya di windows, pakai docker desktop. Kenapa kerasa lambat ya bang ? bener ya spefikasi PC mempengaruhi ?

**Reply by deaafrizal** [reply]
hooh bro emang kalau pakenya yg desktop rada berat. enakan yang CLI aja bro pake wsl kalo windows

**Reply by mwafa** [reply]
docker sangat membantu gw setelah urus banyak apps, dulu pakai pm2 sekarang pindah ke docker-swarm.
